snapd.go

```diff
@@ -645,9 +653,37 @@ func setDurationVal(field time.Duration, ctx *cli.Context, flagName string) time
 	return field
 }
 
+//
+func checkCmdLineFlags(ctx *cli.Context) error {
+	// Bind address is specified
+	if ctx.IsSet("api-addr") {
+		addr := ctx.String("api-addr")
+		// Contains a comma
+		if strings.Index(addr, ",") != -1 {
+			return serror.ErrBadAddress
+		}
+		idx := strings.Index(addr, ":")
+		// Port is specified in address string
+		if idx != -1 {
+			// Port is also specified on command line
+			if ctx.IsSet("api-port") {
+				return errors.New("Port can not be specified in both --api-addr and --port")
+			}
+			if idx == (len(addr) - 1) {
+				return errors.New("Empty port specified")
+			}
+		}
+	}
+	return nil
+}
+
 // Apply the command line flags set (if any) to override the values
 // in the input configuration
 func applyCmdLineFlags(cfg *Config, ctx *cli.Context) {
+	err := checkCmdLineFlags(ctx)
+	if err != nil {
+		panic(err)
```

Comment:
Let's remove the panic and call log.Fatal.  Calling `log.Fatal` calls `os.Exit(1)` after printing the log message.